code/main.py

```diff
@@ -110,6 +110,7 @@
             field_data[form_code_field] = grid_r.read_field_as_string(
                 form_code_field, grid, threshold, form_variant,
                 field_fill_percents[form_code_field]) or ""
+            field_data[grid_i.Field.IMAGE_FILE] = image_path.name
```

Comment:
Since we are always setting `field_data[grid_i.Field.IMAGE_FILE]`, why not delete this line and line 123 and instead just add it in the `field_data` initialization in line 102, like this:
```py
field_data: : tp.Dict[grid_i.RealOrVirtualField, str] = {
  grid_i.Field.IMAGE_FILE: image_path.name
}